library/core/src/any.rs

```diff
@@ -464,6 +464,13 @@ impl TypeId {
     }
 }
 
+#[stable(feature = "rust1", since = "1.0.0")]
+impl fmt::Debug for TypeId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.pad("TypeId(_)")
```

Comment:
`Eq` is implemented for `TypeId`, so it is possible to distinguish two type id's from code, but this will make it impossible to distinguish it in the debug output. In addition `Hash` is derived, making it possible to retrieve the inner integer using a custom hasher.